### PR TITLE
chore: use Getter in WarnOfExpensiveReadOnlyTxnRequest 

### PR DESCRIPTION
**Pull request by fuweid**
The pb provides an accessor method to get field and it will not panic if
the owner is nil. And add non-empty RangeRespone into the test case.

Signed-off-by: Wei Fu <fuweid89@gmail.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

ping @ahrtr

### CODE REVIEW

File: server/etcdserver/txn/util.go [hunk 1]
```diff
@@ -61,13 +61,9 @@ func WarnOfExpensiveReadOnlyTxnRequest(lg *zap.Logger, warningApplyDuration time
...
 		for _, r := range txnResponse.Responses {
-			switch op := r.Response.(type) {
+			switch r.Response.(type) {
 			case *pb.ResponseOp_ResponseRange:
-				if op.ResponseRange != nil {
-					resps = append(resps, fmt.Sprintf("range_response_count:%d", len(op.ResponseRange.Kvs)))
-				} else {
-					resps = append(resps, "range_response:nil")
-				}
+				resps = append(resps, fmt.Sprintf("range_response_count:%d", len(r.GetResponseRange().GetKvs())))
```
Comment (ahrtr) [inline server/etcdserver/txn/util.go:66]: It seems `range_response:nil` is a little better than `range_response:0` in this case. We can easily tell the TXN terminates/fails at which request. 

For example, there are 5 read requests in a TXN, it's easy to tell the TXN terminates/fails at the 3rd request, 
```
request 1:  "range_response_count: 10"
request 2:  "range_response_count: 0" (Note: no any error yet)
request 3:  "range_response_count: nil", some error occurs
request 4:  "range_response_count: nil" (will not be executed at all)
request 5:  "range_response_count: nil" (will not be executed at all)
```

But it's hard to tell which request the TXN terminates/fails at in this PR, 
```
request 1:  "range_response_count: 10"
request 2:  "range_response_count: 0" (Note: no any error yet)
request 3:  "range_response_count: 0", some error occurs
request 4:  "range_response_count: 0" (will not be executed at all)
request 5:  "range_response_count: 0" (will not be executed at all)
```

Comment (ptabor) [reply server/etcdserver/txn/util.go:66]: No strong opinion: 
 1.  Usage of `Get` and `Has` is definitely more aligned with the long-term plans of evolution of protobuf library
     (As getter allow for more eficienient representation than the 'unpacked' properties). So I'm all for preferring this stale of accessing protos. 
     
 2. Agree that the old format caries more information. If the goal is readability, we should pass the underling error as well (but it's a bigger change). 


How about still having the If - as there is no ternary operation in Golang, but using the getters ?

Comment (fuweid) [reply server/etcdserver/txn/util.go]: Thanks for the suggestion. Agree with @ahrtr. I updated it with Getter and keep `range_response:nil`. And also updated the test case with non-empty RangeResponse. PTAL. Thanks!

**Review by ptabor — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @fuweid
